lib/core/src/sync/model/mod.rs

```diff
@@ -0,0 +1,68 @@
+use boltz_client::boltz::SwapTree;
+use serde::{Deserialize, Serialize};
+
+use crate::prelude::Direction;
+
+pub(crate) mod sync;
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct ClaimDetails {
+    pub(crate) swap_tree: SwapTree,
+    pub(crate) timeout_block_height: u32,
+    pub(crate) claim_fees_sat: u64,
+    pub(crate) claim_address: String,
+    pub(crate) claim_private_key: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct LockupDetails {
+    pub(crate) swap_tree: SwapTree,
+    pub(crate) timeout_block_height: u32,
+    pub(crate) lockup_address: String,
+    pub(crate) refund_private_key: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct ChainSyncData {
+    pub(crate) swap_id: String,
+    pub(crate) preimage: String,
+    pub(crate) description: Option<String>,
+    pub(crate) direction: Direction,
+    pub(crate) claim_details: ClaimDetails,
+    pub(crate) lockup_details: LockupDetails,
+    pub(crate) payer_amount_sat: u64,
+    pub(crate) receiver_amount_sat: u64,
+    pub(crate) accept_zero_conf: bool,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct SendSyncData {
+    pub(crate) swap_id: String,
+    pub(crate) preimage: Option<String>,
+    pub(crate) description: Option<String>,
+    pub(crate) refund_private_key: String,
+    pub(crate) swap_tree: SwapTree,
```

Comment:
Also it is better not to rely on external structures on sync. They can change and we don't control that.

Reply:
Agreed, will copy those on our side then